nomos-cli/src/da/network/swarm.rs

```diff
@@ -0,0 +1,133 @@
+use std::collections::HashMap;
+use std::fmt;
+use std::marker::PhantomData;
+
+// std
+// crates
+use kzgrs_backend::common::blob::DaBlob;
+use kzgrs_backend::encoder::EncodedData as KzgEncodedData;
+use libp2p::futures::StreamExt;
+use libp2p::Multiaddr;
+use libp2p::{identity::Keypair, swarm::SwarmEvent, PeerId, Swarm};
+use nomos_core::da::{BlobId, DaDispersal};
+use nomos_da_network_core::protocols::dispersal::executor::behaviour::{
+    DispersalError, DispersalExecutorEvent,
+};
+use nomos_da_network_core::{
+    protocols::dispersal::executor::behaviour::DispersalExecutorBehaviour, SubnetworkId,
+};
+use subnetworks_assignations::MembershipHandler;
+use tokio::sync::mpsc::UnboundedSender;
+use tracing::{debug, error};
+// internal
+
+#[derive(Debug)]
+pub enum DispersalEvent {
+    /// A blob successfully arrived its destination
+    DispersalSuccess {
+        blob_id: BlobId,
+        subnetwork_id: SubnetworkId,
+    },
+    /// Something went wrong delivering the blob
+    DispersalError { error: DispersalError },
+}
```

Comment:
You could probably reuse this type in the Backend directly as they are exactly the same?